### ISSUE TITLE
Price attribute comparison options not showing as expected

### ISSUE THREAD

**Issue by gjportegies** [Smile-SA/elasticsuite#1239]
### Preconditions

Magento Version : 2.2.6 CE

ElasticSuite Version : 2.6.4

Environment : Developer

### Question

In our Magento webshop we have two attributes margin_relative and margin_absolute. The attribute type of those attributes is 'price'. For both attributes I've enabled 'Use in Search' so the attributes are visible when creating a virtual category rule. But when choosing one of the attributes it only gives me two comparison options; _contains_ and _not contains_ while for a 'price' attribute I'm expecting the numeric comparison options ('==', '!=', '>=', '>', '<=', '<').

The getInputType() method defined in the \Smile\ElasticsuiteCatalogRule\Model\Rule\Condition\Product class does contain a check for price but only check if the attribute equals price and not if the type equals price?

When I manually add the following code to the above class the comparisons are working as I would expect.
```
} elseif ($frontendInput === 'price') {
    $inputType = 'numeric';
}
```
But I really would like to know if this is the way to go? And why this is not working by default? Are there any cons of doing this?

**Comment by rbayet** [comment]
Hello @gjportegies,

The test on the attribute_code of `price` has been there since the Virtual Categories were first implemented.
I could not tell you why this is not working by default, there might be an historical reason that is no longer relevant.

Testing on a 2.2.6 CE with Luma, I don't see any problem with indeed allowing attributes like 'cost' and 'msrp' from being used as numeric input into category/optimizers rules.
Those attributes are not subject to tiers pricing (variation by website + customer group) as `price` is and the simple form under which they are indexed.
I haven't tested on a multiple website / multiple currencies instance though, so I'll discuss the matter with @romainruaud (we might end up accepting a PR from you for that).

About "is this the way to go ?" : as `\Smile\ElasticsuiteCatalogRule\Model\Rule\Condition\Product::getInputType` and `\Magento\Rule\Model\Condition\Product\AbstractProduct::getAttributeObject` are public, the proper way would be to write a Plugin for you to avoid a composer patch.

PS: You should also check that this works correctly in the context of Search Optimizers.

Regards,
Richard

**Comment by gjportegies** [comment]
Thanks for quick answer @rbayet ! I do not want to change any code in the vendor directory and keep as less plugins as needed. So would like to help fixing it with a PR. I'll start testing it in the context Search Optimizers while waiting on the answer of @romainruaud 

Thanks!

**Comment by romainruaud** [comment]
Well that's probably a mistake by our side.

I do not see actually any counter-arguments to adding attributes having `$frontendInput === 'price'` being rendered as `numeric`.

Just one point, maybe it could be worth it to also check the `backend_type` of the attributes to display those having `backend_type=decimal` to be rendered as numerics.

I let you propose a PR if you want to @gjportegies 

Thank you for pointing this to us.

Best regards

**Comment by gjportegies** [comment]
I'll create a PR asap!

**Comment by romainruaud** [comment]
Fixed by associated PR, will be part of next minor releases 2.6.7 & 2.7.5

**Comment by gjportegies** [comment]
@romainruaud the solution with ```in_array($frontendClass, ['validate-digits', 'validate-number'])``` didn't work for my attributes.

Although if I check for the `backend_type` like you mentioned it works for my attributes.

I can make a pull request with the following additions (if you agree?):

```$backendType = $subject->getAttributeObject()->getBackendType();```

```if ($backendType === 'decimal') $inputType = 'numeric';```

**Comment by gjportegies** [comment]
Should this issue be reopened @romainruaud @rbayet  ?
